pkg/ebpf/tracer.go

```diff
@@ -77,7 +76,7 @@ type Tracer struct {
 	bufferLock sync.Mutex
 
 	// Internal buffer used to compute bytekeys
-	buf *bytes.Buffer
+	buf [network.ConnectionByteKeyMaxLen]byte
```

Comment:
Why does this make a difference? Either way, `buf` should only be allocated once (or twice for a bytes.Buffer) during the life of the tracer ?

Reply:
Using an array allows us to get rid of allocating an 8 byte array for every bytekey: https://github.com/DataDog/datadog-agent/pull/6709/files#diff-343f83a02a18a428531c7fc955c3603ab42df34e7a6d28d70bc99641436f2571L174